pytorch_lightning/callbacks/quantization.py

```diff
@@ -311,3 +334,19 @@ def on_predict_start(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule
     def on_predict_end(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -> None:
         if "predict" in self._observer_disabled_stages:
             self._restore_last_observer_enabled()
+
+    def state_dict(self) -> Dict[str, Any]:
+        arg_names = ("qconfig", "observer_type", "collect_quantization", "modules_to_fuse", "input_compatible")
+        attribs = {n: getattr(self, f"_{n}") for n in arg_names}
+        return attribs
+
+    def load_before_model(self, pl_module: "pl.LightningModule", state_dict: Dict[str, Any]) -> None:
+        """Special hook that gets called by the CheckpointConnector *before* the model gets loaded.
+
+        This hook replaces the :meth:`on_load_checkpoint` and :meth:`load_state_dict` callback methods which get called
+        after the model has already loaded the weights. For quantization, we need to convert the model first before
+        that happens, assuming the previous training used quantization.
```

Comment:
Check that my explanation is correct. 
Also, note in the last sentence the assumption we make.